### ISSUE TITLE
`git machete github checkout-prs` cannot check out branches from yet-unadded forks

### ISSUE THREAD

**Issue by PawelLipski** [PawelLipski/git-machete#242]
`Creating out of develop` is the problem here :/

```
➜ 21:31 ~/git-machete improve/parser PR #229 (BartoszSiwinski) $ git machete github checkout-prs 241
Fetching origin...
remote: Enumerating objects: 15, done.
remote: Counting objects: 100% (15/15), done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 12 (delta 9), reused 9 (delta 6), pack-reused 0
Unpacking objects: 100% (12/12), 1.32 KiB | 79.00 KiB/s, done.
From github.com:VirtusLab/git-machete
   f88bbc7..8a8b25b  improve/parser -> origin/improve/parser
A local branch feature/add-mypy-env-to-tox does not exist. Creating out of develop
Switched to a new branch 'feature/add-mypy-env-to-tox'
Added branch feature/add-mypy-env-to-tox onto develop
Annotating feature/add-mypy-env-to-tox as PR #241 (ronnie-llamado)
Pull request #241 checked out at local branch feature/add-mypy-env-to-tox
```

**Comment by PawelLipski** [comment]
Maybe checking out the `refs/pulls` remote branches wasn't that bad idea after all 🤔 to be weighed against the alternative (just adding the fork as yet another remote)

**Comment by rifel123** [comment]
Adding forked repo as another remote, may end up having a lot of remotes, which probably leads to  much longer execution time when `fetch` is done for every remote, like it is in `traverse` for example. On the other hand it might be difficult to determine base branch for `refs/pull`. Maybe add branch taken with `refs/pull` as new root?

**Comment by PawelLipski** [comment]
Huh good question :goncern: Actually, for clarity: _ it might be difficult to determine base branch for `refs/pull`_ what do you mean specifically? 🤔

**Comment by rifel123** [comment]
Like for PR #73 there would no base ATM, same thing may happen for closed PRs from forks.

**Comment by PawelLipski** [comment]
Aww okay, the same problem you've mentioned on PR... I think it's unrelated to either handling forks or even whether we use `refs/pulls` or regular remote branches, right? I think it merits a separate issue, in fact

**Comment by rifel123** [comment]
closed with #254
